nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,446 @@
+          };
+          logToStdout = mkOption {
+            type = types.bool;
+            default = true;
+            description = mdDoc "Whether to log to stdout (will go to systemd service). Level can be changed with logLevel.";
+          };
+          blockDailyCheck = mkOption {
+            type = types.bool;
+            default = false;
+            description = mdDoc "Disable telemetry. This has priority over the 'Server telemetry' setting in admin panel.";
+          };
+          installMethod = mkOption {
+            type = types.str;
+            default = "nixos";
+            description = mdDoc ''
+              Install method is listed in telemetry if you agree to it through the consentToContact
+              setting in the admin panel.
+            '';
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      users.users.cryptpad = {
+        isSystemUser = true;
+        group = "cryptpad";
+      };
+      users.groups.cryptpad = {};
+
+      systemd.services.cryptpad = {
+        description = "Cryptpad service";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "networking.target" ];
+        serviceConfig = {
+          User = "cryptpad";
+          Environment = [
+            "CRYPTPAD_CONFIG=${conf_file}"
+            "HOME=%S/cryptpad"
+          ];
+          ExecStart = "${cfg.package}/bin/cryptpad";
+          PrivateTmp = true;
+          Restart = "always";
+          StateDirectory = "cryptpad";
+          WorkingDirectory = "%S/cryptpad";
+        };
+      };
+    }
+    (mkIf cfg.confinement {
+      systemd.services.cryptpad = {
+        serviceConfig.BindReadOnlyPaths = [
+          conf_file
+          # apparently needs proc for workers management
+          "/proc"
+          "/dev/urandom"
+        ] ++ (if ! cfg.config.blockDailyCheck then [
+          "/etc/resolv.conf"
+          "/etc/hosts"
+        ] else []);
+        confinement = {
+          enable = true;
+          binSh = null;
+          mode = "chroot-only";
+        };
+      };
+    })
+    (mkIf cfg.nginx.enable {
+      assertions = [
+        { assertion = nginx.enable;
+          message = "nginx must be enabled";
+        }
+        { assertion = lib.strings.hasPrefix "https://" cfg.config.httpUnsafeOrigin;
+          message = "services.cryptpad.config.httpUnsafeOrigin must start with https://";
+        }
+        { assertion = cfg.config.httpSafeOrigin == null  || lib.strings.hasPrefix "https://" cfg.config.httpSafeOrigin;
+          message = "services.cryptpad.config.httpSafeOrigin must start with https:// (or be unset)";
+        }
+      ];
+      services.nginx.virtualHosts = mkMerge [
+        {
+          ${nginx_main_domain} = {
+            forceSSL = lib.mkDefault true;
+            # Probably not wise to automatically enable acme?...
+            enableACME = lib.mkDefault config.security.acme.acceptTerms;
```

Comment:
left a comment in the code here but not quite sure what to do about ACME. Checking if acceptTerms seems safe enough to do the right thing, but it might be a bit surprising... jitsi for example just sets enableACME to `mkDefault true` so this is slightly better?

Reply:
Not really, no, I'm not aware of module doing this by default, some people may accept terms and did not set the DNS record yet, so please, just let the user enable ACME explicitly.